Tensile/SolutionStructs.py

```diff
@@ -2948,6 +2948,55 @@ def assignDerivedParameters(state):
         if state["LocalSplitU"] > 1:
           reject(state, "Atomic Stream-K not working with LocalSplitU")
 
+    if state["KernelLanguage"] == "Assembly" and not state["BufferLoad"]:
+      # StaggerU only works with source kernels, or with BufferLoad.
+      # Since StaggerU defaults to 32, override it to 0 if not supported.
+      state["StaggerU"] = 0
+
+    def supportsPreloadKernelArguments():
+      if not globalParameters["AsmCaps"][isa]["KernargPreloading"]:
+        return False, f"{isa} doesn't support preloading."
+      
+      if not state["ProblemType"]["StridedBatched"]:
+        return False, "Preloading only supported in StridedBatched mode."
+
+      dt = state["ProblemType"]["DataType"]
+      if not (dt.isSingle() or dt.isHalf() or dt.isBFloat16()):
+        return False, f"Preloading not supported for data type {dt}."
+
+      if not state["ProblemType"]["UseBeta"]:
+        return False, "Preloading only supported with UseBeta."
+
+      if state["ProblemType"]["UseInitialStridesAB"]:
+        return False, "Preloading not supported with initial strides AB."
+
+      if abs(state["WorkGroupMapping"]) > 1:
+        return False, "Preloading not supported with WorkGroupMapping."
+
+      if not state["PrefetchGlobalRead"]:
+        return False, "Preloading must use PrefetchGlobalRead."
+
+      if state["EdgeType"] == "ShiftPtr":
```

Comment:
Why can't we use preload arg for ShiftPtr?
Is it difficult to support?
We usually use EdgeType=ShiftPtr.

Reply:
ShiftPtr will use some of the remaining tensor sizes in the calculations to shift the pointers before the first load of A and B.  There's not enough space for those sizes in the preloaded args.

It's possible we could defer shifting the pointers until after the first load but that would require some effort.

Reply:
OK. 
I will try benchmark with branch.
If the performance is not as good as ShiftPtr, probably, it is better to move global read prefetch only for A side and enable it for more generic cases.